lib/Service/ShareService.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * Nextcloud - News
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author    Marco Nassabain <marco.nassabain@hotmail.com>
+ */
+
+namespace OCA\News\Service;
+
+use \OCA\News\Db\Item;
+use \OCA\News\Db\Feed;
+
+use \Psr\Log\LoggerInterface;
+
+use OCA\News\Service\Exceptions\ServiceNotFoundException;
+use OCP\AppFramework\Db\DoesNotExistException;
+
+/**
+ * Class ShareService
+ *
+ * @package OCA\News\Service
+ */
+class ShareService
+{
+    /**
+     * Items service.
+     *
+     * @var ItemServiceV2
+     */
+    protected $itemService;
+
+    /**
+     * Feeds service.
+     *
+     * @var FeedServiceV2
+     */
+    protected $feedService;
+    
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * ShareService constructor
+     *
+     * @param FeedServiceV2   $feedService Service for feeds
+     * @param ItemServiceV2   $itemService Service to manage items
+     * @param LoggerInterface $logger      Logger
+     */
+    public function __construct(
+        FeedServiceV2 $feedService,
+        ItemServiceV2 $itemService,
+        LoggerInterface $logger
+    ) {
+        $this->itemService = $itemService;
+        $this->feedService = $feedService;
+        $this->logger      = $logger;
+    }
+
+    /**
+     * Share an item with a user
+     *
+     * @param string $userId           ID of user sharing the item
+     * @param int    $itemId           Item ID
+     * @param string $shareRecipientId ID of user to share with
+     *
+     * Sharing by copying - the item is duplicated, and the 'sharedBy'
+     * field is filled accordingly.
+     * The item is then placed in a dummy feed reserved for items
+     * shared with the user
+     *
+     * @return Item Shared item
+     * @throws ServiceNotFoundException|ServiceConflictException
+     */
+    public function shareItemWithUser(string $userId, int $itemId, string $shareRecipientId)
+    {
+        // Find item to share
+        $item = $this->itemService->find($userId, $itemId);
+
+        // Duplicate item & initialize fields
+        $sharedItem = clone $item;
+        $sharedItem->setId(null);
+        $sharedItem->setUnread(true);
+        $sharedItem->setStarred(false);
+        $sharedItem->setSharedBy($userId);
+
+        // Get 'shared with me' dummy feed
+        $feedUrl = 'http://nextcloud/sharedwithme';
+        $feed = $this->feedService->findByUrl($shareRecipientId, $feedUrl);
+        if (is_null($feed)) {
+            $feed = new Feed();
+            $feed->setUserId($shareRecipientId)
+                 ->setUrlHash(md5($feedUrl))
+                 ->setLink($feedUrl)
+                 ->setUrl($feedUrl)
+                 ->setTitle('Shared with me')
```

Comment:
It would also vote for [translation support](https://docs.nextcloud.com/server/latest/developer_manual/basics/front-end/l10n.html#php) here (I know the import currently doesn't).

Reply:
Just to be sure, localized as in writing a function in the feedService that fetches/creates a dummy url?
I've thought about doing this, but since I was basing my code on the ImportService, I thought it would be better to follow in your footsteps.
Good idea for the translation support, I will add that in.

Reply:
As in the following, plus the `IL10N` in `__construct`.
```suggestion
                 ->setTitle($this->l->t('Shared with me'))
```

Reply:
Localization is what you call translation support. Since you're not actually changing anything for the locale I guess both are valid: https://lokalise.com/blog/translation-and-localization-difference/

Reply:
Oh yea 🤦‍♂️😅 my bad